socs/agents/hwp_supervisor/agent.py

```diff
@@ -1154,6 +1167,7 @@ def check_gripper_ok_for_spinup():
                         target_freq=state.target_freq,
                         freq_tol=state.freq_tol,
                         freq_tol_duration=state.freq_tol_duration,
+                        max_duration=1800,
```

Comment:
Instead of hardcoding 1800 multiple places, we should define a global constant at the top like:
```
MAX_SPIN_UP_DURATION: float = 1800.
```
which we can check against inside the update function. 

If we're hardcoding this and not expecting it to change, then I don't think we have to include it as a state field.